### ISSUE TITLE
Multi-IdP API: The RP needs a way to know which IdP the user selected

### ISSUE THREAD

**Issue by samuelgoto** [fedidcg/FedCM#560]
> I'm posting this on behalf of an IdP that is currently experimenting with the Multi-IdP API and has given us this feedback

In the [Multi-IdP API](https://github.com/fedidcg/FedCM/issues/319) proposal, the RP needs to know which IdP was selected from the list of [`providers`](https://fedidcg.github.io/FedCM/#dom-identitycredentialrequestoptions-providers), so that it can interpret the [`token`](https://fedidcg.github.io/FedCM/#dom-identitycredential-token) that it gets in the [`IdentityCredential`](https://fedidcg.github.io/FedCM/#identitycredential).

**Comment by samuelgoto** [comment]
One of the proposals that @npm1 is exploring is exposing the [`configURL`](https://fedidcg.github.io/FedCM/#dom-identityproviderconfig-configurl) that was used when the user made the choice in the resulting [`IdentityCredential`](https://fedidcg.github.io/FedCM/#identitycredential).

**Comment by obfuscoder** [comment]
Or the configURL provided by the IdP during IdP registration (see #240) when client used "any" as IdP configURL?

**Comment by samuelgoto** [comment]
> Or the configURL provided by the IdP during IdP registration (see https://github.com/fedidcg/FedCM/issues/240) when client used "any" as IdP configURL?

Yep.

**Comment by obfuscoder** [comment]
I read somewhere else that this is already available in Canary for testing?

**Comment by cbiesinger** [comment]
Yes! @npm1 added a configURL property to the returned IdentityCredential in version 126.0.6436.0 (requires the multi IDP flag to be enabled)

**Comment by samuelgoto** [comment]
> Yes! @npm1 added a configURL property to the returned IdentityCredential in version 126.0.6436.0 (requires the multi IDP flag to be enabled)

This cl [here](https://chromium-review.googlesource.com/c/chromium/src/+/5472945/5/third_party/blink/renderer/modules/credentialmanagement/identity_credential.idl).

As @cbiesinger, you should be able to use the `configURL` in the returning `IdentityCredential` to figure out which IdP that was used that resulted into the promise.

I think @npm1 was intending it to be used somewhat like the following:

```javascript
const credential = await navigator.credentials.get({
  identity: {
    providers: [{
      configURL: "https://idp1.example",
      // ... other stuff ...
    }, {
      configURL: "https://idp2.example",
      // ... other stuff ...
    }, ]
  }
});

const {token, configURL} = credential;

// decode the token differently depending on which configURL was used
```

**Comment by cbiesinger** [comment]
We should probably keep this open until we actually added this to the spec

**Comment by aaronpk** [comment]
Just wanted to chime in and say this is working for me.
